harvester/harvest/alma.py

```diff
@@ -42,13 +42,14 @@ def full_harvest_get_source_records(self) -> Iterator[Record]:
 
         For a full harvest, the harvester will examine the list of XML files
         to determine the latest full run date by parsing dates from the filepaths,
-        which is assigned to MITAlmaHarvester.from_date. This enables the harvester
+        which is then assigned to self.from_date. This enables the harvester
         to get source records the from corresponding "full-extracted" XML file(s) and
         "daily-extracted" XML file(s) with a run date >= latest full run date.
         """
         CONFIG.check_required_env_vars()
         filepaths = self._list_xml_files()
-        self.from_date = self._get_latest_full_run_date(filepaths)
+        if from_date := self._get_latest_full_run_date(filepaths):
+            self.from_date = from_date
```

Comment:
Perhaps this is related to [your comment here](https://github.com/MITLibraries/geo-harvester/pull/271#discussion_r1563007821), but perhaps this would be a good throw an error is `self.from_date` is not set?  Additionally, perhaps we allow `self.from_date` to override if already set?  

Something along the lines of:
```python
if not self.from_date:
    from_date = self._get_latest_full_run_date(filepaths)
if not self.from_date:
    raise Exception("'from_date' required for full alma harvests.")
```

Could play with the syntax some, but the idea is:
1. If `self.from_date` is not already set, extract from listed files
2. If _still_ not set, throw an error, as we shouldn't proceed with full harvest unless set on the harvester

Reply:
Please see the latest commit!